jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -163,14 +223,51 @@ static JarType determineJarType(Path jarPath) throws IOException {
           jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.MAIN_CLASS);
       if (mainClass == null) {
         throw new IllegalArgumentException(
-            "`Main-Class:` attribute for an application main class not defined in the input Jar's "
-                + "manifest (`META-INF/MANIFEST.MF` in the Jar).");
+            "`Main-Class:` attribute for an application main class not defined in the input JAR's "
+                + "manifest (`META-INF/MANIFEST.MF` in the JAR).");
       }
       String classpath = APP_ROOT + "/explodedJar:" + APP_ROOT + "/dependencies/*";
       return ImmutableList.of("java", "-cp", classpath, mainClass);
     }
   }
 
+  /**
+   * Computes the entrypoint for a standard jar in packaged mode.
+   *
+   * @param jarPath path to jar file
+   * @return list of {@link String} representing entrypoint
+   * @throws IOException if I/O error occurs when opening the jar file
+   */
+  static ImmutableList<String> computeEntrypointForPackagedStandard(Path jarPath)
+      throws IOException {
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      String mainClass =
+          jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.MAIN_CLASS);
+      if (mainClass == null) {
+        throw new IllegalArgumentException(
+            "`Main-Class:` attribute for an application main class not defined in the input JAR's "
+                + "manifest (`META-INF/MANIFEST.MF` in the JAR).");
+      }
+      return ImmutableList.of("java", "-jar", APP_ROOT + "/" + jarPath.getFileName().toString());
+    }
+  }
+
+  private static ImmutableMap<String, List<Path>> getSnapshotAndNonSnapshotDependencies(
```

Comment:
I'm not a big fan of using a map for the purpose of returning more than one value. How about creating a private inner class? For this purpose, I think we don't even need getters and setters.

```java
private static class LocalDependencies {
  private final List<Path> snapshots;
  private final List<Path> nonSnapshots;

  private LocalDependencies(List<Path> snapshots, List<Path> nonSnapshots) {
    this.snapshots = snapshots;
    this.nonSnapshots = nonSnapshots;
  }
}
```

But is it not possible for this method to return built layers? (I remember the original code worked in that way, and I liked it.) In that way, I think we can remove duplicate code between the exploded mode and the packaged mode?